### ISSUE TITLE
OS X Integration walkthrough

### ISSUE THREAD

**Issue by alzbookstore** [jasperproject/jasper-client#324]
I am hoping to receive pointers to the correct resolved issues as I am having difficulty finding clarity on OS X installation.

During configuration, I am having several issues that seem to be unrelated (No module named yaml, AttributeErrors, etc.) although I've triple-checked that Python, pyyaml, profile.yml., etc. are all seemingly installed correctly.

Question:
Does anyone have links to any resolved Issues or Commits that apply to OS X installation/configuration (specifically with Google STT, which I've already configured with API key) that gave them a good walkthrough of successful installation and run of Jasper?

I am simply having difficulty finding the right starting point for support.

I apologize if this is much to vague -- I'm happy to elaborate if anyone can point me in the right direction.

Thank you so much in advance!

**Comment by tim-toady** [comment]
I ran into the same problem (no module named yaml), the reason, in my case, was that the installation of yaml was by default installed to Python 2.7, while Jasper was configured to Python 2.6.
The solution was pretty easy to fix. A little change in the jasper.py file at the top:
`#!/usr/bin/env python2`
to
`#!/usr/bin/env python`
fixed it for me.

Still having issues with the STT though (i'm using PocketSphinx); which fails to open an audio device due to an A/D library not being implemented. Ideas are welcome.

**Comment by alzbookstore** [comment]
Fantastic! That resolved the yaml module issue. Interesting that it continued to error although I changed Python default to 2.7. Now, of course, I'm getting another error:

    from client import tts, stt, jasperpath, diagnose
    ImportError: No module named client

I don't see this module in the Python 2.7 library. I wonder if something else should be redirecting that `from` command? Ideas?

Re:PocketSphinx, I'm going straight to Google API for my STT so I don't have much helpful advice there. Are you installing it through homebrew? Have you checked out some steps others have taken from Issue #35?
User @astahlman seems to have had good luck if you've followed his work (shows up about 1/3 way down that page.)

If using homebrew and still having issues, you may check out https://github.com/watsonbox/homebrew-cmu-sphinx, but I don't know how helpful that would be.

Otherwise, I'm sorry I don't have more on that front.

**Comment by tim-toady** [comment]
Client is actually referring to the client directory of Jasper. My guess here is that you didn't run the command while being in the right directory.
I remember seeing something similar in my prompt, but of course I didn't document and can't recall under which circumstances this appeared.

Re: Good call on the watsonbox, I managed to miss that reply. Still having some issues though, but haven't found the time to continue on it yet.

**Comment by alzbookstore** [comment]
That seems accurate. I suppose I confused something because I've got a 'jasper' directory (which contains jaspery.py among other things) and a 'jasper-client' directory. I honestly don't remember how I came upon them, but both contain client and boot folders. I need to figure out how to resolve those...

What is your general directory structure? Does everything run out of the jasper-client?

Re: Some dependencies might also be more easily or successfully installed via pip rather than homebrew. Not sure which though regarding PocketSphinx.

I appreciate your help!

**Comment by tim-toady** [comment]
Yeah, I just have a jasper-client folder containing a client folder and I run everything out of the jasper-client.

**Comment by zoecarver** [comment]
@alzbookstore Let me know if you still need any help getting it to run.

### Edit, here are some instructions: https://github.com/pudility/jasper-osx
